sources/packages/api/src/route-controllers/application/application.aest.controller.ts

```diff
@@ -0,0 +1,148 @@
+  Controller,
+  Get,
+  NotFoundException,
+  Param,
+  Query,
+} from "@nestjs/common";
+import {
+  ApplicationService,
+  EducationProgramOfferingService,
+  InstitutionLocationService,
+  EducationProgramService,
+} from "../../services";
+import BaseController from "../BaseController";
+import {
+  GetApplicationBaseDTO,
+  transformToApplicationDto,
+  StudentApplicationAndCount,
+  ApplicationFormData,
+} from "./models/application.model";
+import { AllowAuthorizedParty, Groups } from "../../auth/decorators";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { UserGroups } from "../../auth/user-groups.enum";
+import { Application } from "../../database/entities";
+import {
+  FieldSortOrder,
+  DEFAULT_PAGE_NUMBER,
+  DEFAULT_PAGE_LIMIT,
+  transformToApplicationSummaryDTO,
+  getOfferingNameAndPeriod,
+} from "../../utilities";
+import { ApiNotFoundResponse, ApiOkResponse, ApiTags } from "@nestjs/swagger";
+
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("application")
+@ApiTags("application")
+export class ApplicationAESTController extends BaseController {
+  constructor(
+    private readonly applicationService: ApplicationService,
+    private readonly offeringService: EducationProgramOfferingService,
+    private readonly locationService: InstitutionLocationService,
+    private readonly programService: EducationProgramService,
+  ) {
+    super();
+  }
+
+  /**
+   * API to fetch application details by applicationId.
+   * This API will be used by ministry users.
+   * @param applicationId
+   * @returns Application details
+   */
+  @Get(":applicationId")
+  @ApiOkResponse({ description: "Application details fetched." })
+  @ApiNotFoundResponse({ description: "Application not found." })
+  async getApplication(
+    @Param("applicationId") applicationId: number,
+  ): Promise<GetApplicationBaseDTO> {
+    const application = await this.applicationService.getApplicationByIdAndUser(
+      applicationId,
+    );
+    if (!application) {
+      throw new NotFoundException(
+        `Application id ${applicationId} was not found.`,
+      );
+    }
+    const additionalFormData = {} as ApplicationFormData;
+    // Check wether the selected location is designated or not.
+    // If selected location is not designated, then make the
+    // selectedLocation null
+    if (application.data.selectedLocation) {
+      const designatedLocation =
+        await this.locationService.getDesignatedLocationById(
+          application.data.selectedLocation,
+        );
+      const selectedLocation = await this.locationService.getLocationById(
+        application.data.selectedLocation,
+      );
+      if (!designatedLocation) {
+        application.data.selectedLocation = null;
+      }
+      // Assign location name for readonly form
+      additionalFormData.selectedLocationName = selectedLocation.name;
+    }
+    // Check wether the program is approved or not.
```

Comment:
The comment says that  checking whether the program is approved or not. but the code is not using approval status.

Reply:
updating comment